<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="cv_style.css" type="text/css" />
</head>
<body>
<!--  

Great resource https://blog.chmd.fr/editing-a-cv-in-markdown-with-pandoc.html 

The CSS is attributable to the above blog post

This is another great post influencing this document https://mszep.github.io/pandoc_resume/

see github repo for licensing information
-->
<h1 id="nicholas-giangreco"><a href="http://systemsbiology.columbia.edu/people/nicholas-giangreco">Nicholas Giangreco</a></h1>
<table>
<tbody>
<tr class="odd">
<td align="left"><u><strong>Email:</strong></u></td>
<td align="right">nick.giangreco@gmail.com</td>
</tr>
<tr class="even">
<td align="left"><u><strong>Links:</strong></u></td>
<td align="right"><a href="http://github.com/ngiangre">Github</a> <a href="http://www.linkedin.com/in/nickgiangreco/">Linkedin</a> <a href="https://orcid.org/0000-0001-8138-4947">ORCID</a></td>
</tr>
<tr class="odd">
<td align="left"><u><strong>Website:</strong></u></td>
<td align="right"><a href="http://nickg.bio">nickg.bio</a></td>
</tr>
<tr class="even">
<td align="left"><u><strong>Programming Languages:</strong></u></td>
<td align="right">Frequent: R, Python, Bash, SQL</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="right">Previously: Java</td>
</tr>
<tr class="even">
<td align="left"><u><strong>Web design:</strong></u></td>
<td align="right">HTML, CSS, Bootstrap library</td>
</tr>
<tr class="odd">
<td align="left"><u><strong>Formatting:</strong></u></td>
<td align="right">LaTeX, Markdown, Jupyter Notebooks, Rmarkdown, Microsoft Office Suite</td>
</tr>
</tbody>
</table>
<h2 id="education">EDUCATION</h2>
<dl>
<dt>2021 (expected)</dt>
<dd><p><strong>PhD Candidate, Systems Biology</strong>; Columbia University, New York City, NY</p>
<p>PhD advisor: <a href="http://tatonettilab.org">Nicholas Tatonetti</a></p>
<p>Topics: Translational Bioinformatics, Pharmacoepidemiology, Analysis of Electronic Health Records</p>
</dd>
<dt>2010-2014</dt>
<dd><p><strong>BS, Biochemistry</strong>; University of Rochester, Rochester, NY</p>
<p>Minor: Philosophy</p>
</dd>
</dl>
<h2 id="work-experience">WORK EXPERIENCE</h2>
<dl>
<dt>2014-Present</dt>
<dd><p><strong>National Human Genome Research Institute</strong>; Bethesda, MD</p>
<p>Principal Investigator: <a href="https://www.genome.gov/12514761/elnitski--group/">Dr. Laura Elnitski</a></p>
<ul>
<li><p>Molecular events during tumorigenesis of murine ovarian tumor subtypes differing in deletion of tumor suppressor genes</p></li>
<li><p>RNA sequencing, DNA methylation sequencing, and their integration.</p></li>
</ul>
<p>Trainee 2014-2016</p>
<p>Special Volunteer 2016-Present</p>
</dd>
</dl>
<h2 id="fellowships-and-awards">FELLOWSHIPS AND AWARDS</h2>
<ul>
<li><p>Columbia Diversity Fellowship</p></li>
<li><p>Department of Systems Biology Merit Fellowship</p></li>
<li><p>Donald Charles Award, University of Rochester Department of Biology</p></li>
<li><p>Fulbright Fellowship Alternate 2013-2014: Sweden, Molecular Modeling, “Novel Antibody-SpA Complex Modeling”.</p></li>
<li><p>Travel Award to 9<sup>th</sup> Student Council and ISMB/ECCB conference 2013 Berlin, Germany.</p></li>
</ul>
<h2 id="publications-and-posters">PUBLICATIONS AND POSTERS</h2>
<ul>
<li><strong>Giangreco N</strong>, Zorn E, Chen E et al. Identification of novel primary graft dysfunction biomarkers using exosome proteomics [version 1; not peer reviewed]. F1000Research 2017, 6:2080 <a href="https://f1000research.com/posters/6-2080">(poster)</a> (doi: 10.7490/f1000research.1115115.1)</li>
<li>Sarah Kim-Hellmuth, Matthias Bechheim, Benno Puetz, Pejman Mohammadi, Yohann Nedelec, <strong>Nicholas Giangreco</strong>, Jessica Becker, Vera Kaiser, Nadine Fricker, Esther Beier, Peter Boor, Stephane Castel, Markus M. Noethen, Luis B. Barreiro, Joseph K. Pickrell, Bertram Mueller-Myhsok, Tuuli Lappalainen, Johannes Schumacher, Veit Hornung. Genetic regulatory effects modified by immune activation contribute to autoimmune disease associations <a href="https://www.nature.com/articles/s41467-017-00366-1"><em>Nature Communications</em>, 8 (266): 1-10</a>.</li>
<li><strong>Giangreco N.</strong>, Petrykowska H., Scott A., Margolin G., Gotea V., Cho K. R., and Elnitski L. Inactivation of <em>Arid1a</em> drives aberrant epigenetic traits in a mouse model of <em>Apc</em>/<em>Pten</em> defective ovarian endometrioid tumors. <em>(in preparation)</em></li>
<li><strong>Giangreco N</strong> and Lezon T. Alternative conformation prediction of Vibrio Cholerae concentrative nucleoside transporter. F1000Posters 2013, 4:776 <a href="https://f1000research.com/posters/1093921">(poster)</a>.</li>
</ul>
<h2 id="conferences-and-hackathons">CONFERENCES AND HACKATHONS</h2>
<ul>
<li>American Heart Association Scientific Sessions 2017, poster presentation <a href="https://f1000research.com/posters/6-2080"><em>Giangreco et al. 2017</em></a>.</li>
<li><a href="https://github.com/NCBI-Hackathons/Proteomic_Correlation_Shiny">NCBI Hackathon @ New York Genome Center June 2017.</a></li>
<li><a href="https://github.com/NCBI-Hackathons/Scan2CNV">NCBI Hackathon @ NCBI March 2017.</a></li>
<li>CSHL Biological Data Science meeting October 2016.</li>
<li><a href="https://github.com/NCBI-Hackathons/DASH_cell_type">JHU DaSH Hackathon September 2015.</a></li>
<li>ISMB/ECCB conference @ Berlin, Germany July 2013, poster presentation <em>Giangreco et al. 2013</em>.</li>
</ul>
<h2 id="talks-and-panels">TALKS AND PANELS</h2>
<ul>
<li><a href="http://nickg.bio/_posts/20171220_NYHAIS_talk.pdf">&quot;Tools, Libraries and Analyses in Biomedical Data Science&quot;</a>, New York Healthcare Artificial Intelligence Society, December 2017.</li>
<li><a href="http://nickg.bio/_posts/20171205_LateNightScience_Talk.pdf">&quot;Doing Science with Big Data&quot;</a>, Late Night Science, Columbia University Neuroscience Outreach, December 2017</li>
<li>&quot;AI, Life Sciences, and Big Data&quot;, New York Healthcare Artificial Intelligence Society, August 2017</li>
<li>NIDDK Undergraduate Step-Up Judge, NIH, Bethesda MD, August 2015</li>
</ul>
<h2 id="professional-memberships">PROFESSIONAL MEMBERSHIPS</h2>
<ul>
<li>American Heart Association, 2017-</li>
<li>International Society of Computational Biology, 2013-2014 &amp; 2017-</li>
</ul>
<h2 id="community-involvement">COMMUNITY INVOLVEMENT</h2>
<ul>
<li>NYC MeetUps
<ul>
<li><a href="https://www.meetup.com/AI-at-CUMC/">AI @ Columbia University Medical Center</a>
<ul>
<li>Attend and engage in public discourse on a wide range of topics such as AI &amp; Society, AI &amp; Healthcare, and economic impact by AI.</li>
<li>Facilitate group engagement and organization by consult</li>
</ul></li>
<li><a href="https://www.meetup.com/Data-Storytelling-NYC/">Data Storytelling</a>
<ul>
<li>Monthly lectures by data scientists and managers from private companies presenting stories and narratives using visualizations and interactive methods on diverse sources of data.</li>
</ul></li>
<li><a href="https://www.meetup.com/nyhackr/">Statistical programming</a>
<ul>
<li>Monthly seminars by data scientists, engineers and analysts using novel and statistically rigorous methods applied within academic and company research teams mainly using the R programming language.</li>
</ul></li>
</ul></li>
<li><a href="https://github.com/ngiangre/CUMC_Data_Science_Group">CUMC Data Science Study Group</a>
<ul>
<li>Co-organizer of mostly wet-lab biologists learning data science topics using the R programming language</li>
<li>Leading project management for bioinformatics projects.</li>
</ul></li>
</ul>
</body>
</html>
