<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>Nicholas Giangreco</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="cv_style.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header>
<h1 class="title">Nicholas Giangreco</h1>
</header>
<!--  

Great resource https://blog.chmd.fr/editing-a-cv-in-markdown-with-pandoc.html 

The CSS is attributable to the above blog post

This is another great post influencing this document https://mszep.github.io/pandoc_resume/

see github repo for licensing information
-->
<table>
<colgroup>
<col style="width: 44%" />
<col style="width: 55%" />
</colgroup>
<thead>
<tr class="header">
<th><u><strong>Date of preparation:</strong></u> December 25th 2019</th>
<th style="text-align: right;"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><u><strong>Email Address:</strong></u></td>
<td style="text-align: right;">nick.giangreco@gmail.com</td>
</tr>
<tr class="even">
<td><u><strong>Formatting:</strong></u></td>
<td style="text-align: right;">Markdown, Rmarkdown, Jupyter Notebooks</td>
</tr>
<tr class="odd">
<td><u><strong>Programming Languages:</strong></u></td>
<td style="text-align: right;">Python, R, Bash, SQL, Java</td>
</tr>
<tr class="even">
<td><u><strong>Programming Techniques:</strong></u></td>
<td style="text-align: right;">Parallelization, Out-of-core learning, Distributed computing</td>
</tr>
<tr class="odd">
<td><u><strong>Programming Platforms:</strong></u></td>
<td style="text-align: right;">Multicore machines, Amazon services (e.g. EC2), Databricks</td>
</tr>
<tr class="even">
<td><u><strong>Project Management:</strong></u></td>
<td style="text-align: right;">Teamwork, Slack, Asana, Basecamp, Gitter, DropBox, Google</td>
</tr>
<tr class="odd">
<td><u><strong>Web design:</strong></u></td>
<td style="text-align: right;">Plotly (Dash &amp; Shiny), HTML, CSS, Bootstrap</td>
</tr>
<tr class="even">
<td><u><strong>Web links:</strong></u></td>
<td style="text-align: right;"><a href="http://github.com/ngiangre">Github</a> <a href="http://www.linkedin.com/in/nickgiangreco/">Linkedin</a> <a href="https://orcid.org/0000-0001-8138-4947">ORCID</a> <a href="http://nickg.bio">nickg.bio</a></td>
</tr>
</tbody>
</table>
<h2 id="education">EDUCATION</h2>
<dl>
<dt>August 2016 - Present</dt>
<dd><p><a href="http://systemsbiology.columbia.edu/people/nicholas-giangreco"><strong>PhD Candidate, Systems Biology</strong></a>; Columbia University, New York City, NY</p>
<p>PhD advisor: <a href="http://tatonettilab.org">Nicholas Tatonetti</a></p>
<p>Masters of Arts (2018) and Masters of Philosophy (2019)</p>
</dd>
<dt>August 2010 - May 2014</dt>
<dd><strong>BS, Biochemistry</strong>; University of Rochester, Rochester, NY
</dd>
</dl>
<h2 id="work-experience">WORK EXPERIENCE</h2>
<dl>
<dt>June 2019 - August 2019</dt>
<dd><p><a href="https://www.regeneron.com/genetics-center"><strong>Regeneron Genetics Center</strong></a>; Tarrytown, NY</p>
<p>Clinical informatics intern</p>
</dd>
<dt>July 2018 - September 2018</dt>
<dd><p><a href="https://www.geneticintelligence.com/"><strong>Genetic Intelligence Inc.</strong></a>; New York, NY</p>
<p>Computational biology intern</p>
</dd>
<dt>August 2014-May 2019</dt>
<dd><p><strong>National Human Genome Research Institute</strong>; Bethesda, MD</p>
<p>Principal Investigator: <a href="https://www.genome.gov/12514761/elnitski--group/">Dr. Laura Elnitski</a></p>
<p>Cancer bioinformatics trainee 2014-2016</p>
<p>Special volunteer 2016-2019</p>
</dd>
</dl>
<h2 id="fellowships-and-awards">FELLOWSHIPS AND AWARDS</h2>
<ul>
<li><p>Three-Minute Thesis finalist @ Columbia Graduate School of Arts and Sciences</p></li>
<li><p>Best contribution in methodological research at the <a href="https://www.ohdsi.org/2018-ohdsi-symposium/">OHDSI 2018 Symposium</a> for <a href="http://10.7490/f1000research.1116203.1">Pediatric Drug Safety</a> poster.</p></li>
<li><p>Columbia Diversity Fellowship.</p></li>
<li><p>Department of Systems Biology Merit Fellowship.</p></li>
<li><p>Donald Charles Award, University of Rochester Department of Biology.</p></li>
<li><p>Fulbright Fellowship Alternate 2013-2014: Sweden, Molecular Modeling, “Novel Antibody-SpA Complex Modeling”.</p></li>
<li><p>Travel Award to 9<sup>th</sup> Student Council and ISMB/ECCB conference 2013 Berlin, Germany.</p></li>
</ul>
<h2 id="publications">PUBLICATIONS</h2>
<ul>
<li><strong>Nicholas P. Giangreco</strong>, Barry Fine, Nicholas P. Tatonetti. cohorts: A Python package for clinical ‘omics data management. <a href="https://www.biorxiv.org/content/10.1101/626051v1.article-info">bioarxiv</a></li>
<li>Benjamin S Glicksberg, Boris Oskotsky, Phyllis M Thangaraj, <strong>Nicholas Giangreco</strong>, Marcus A Badgeley, Kipp W Johnson, Debajyoti Datta, Vivek A Rudrapatna, Nadav Rappoport, Mark M Shervey, Riccardo Miotto, Theodore C Goldstein, Eugenia Rutenberg, Remi Frazier, Nelson Lee, Sharat Israni, Rick Larsen, Bethany Percha, Li Li, Joel T Dudley, Nicholas P Tatonetti, Atul J Butte, PatientExploreR: an extensible application for dynamic visualization of patient clinical history from electronic health records in the OMOP common data model, Bioinformatics, , btz409, <a href="https://doi.org/10.1093/bioinformatics/btz409" class="uri">https://doi.org/10.1093/bioinformatics/btz409</a></li>
<li>Benjamin S Glicksberg, Boris Oskotsky, <strong>Nicholas Giangreco</strong>, Phyllis M Thangaraj, Vivek Rudrapatna, Debajyoti Datta, Remi Frazier, Nelson Lee, Rick Larsen, Nicholas P Tatonetti, Atul J Butte, ROMOP: a light-weight R package for interfacing with OMOP-formatted electronic health record data, JAMIA Open, Volume 2, Issue 1, April 2019, Pages 10–14, <a href="https://doi.org/10.1093/jamiaopen/ooy059" class="uri">https://doi.org/10.1093/jamiaopen/ooy059</a></li>
<li>Castillero E., Ali Z., Akashi H., <strong>Giangreco N.</strong>, Wang C., Ji R., Zhang X., Kheysin N., Park J., Hegde S., Patel S., Stein S., Cuenca C., Leung D., Homma S., Tatonetti N., Topkara V., Takeda K., Colombo P., Naka Y., Sweeny L., Schulze C., George I. Structural and Functional Cardiac Profile after Prolonged Duration of Mechanical Unloading: Potential Implications for Myocardial Recovery. <a href="https://www.physiology.org/doi/10.1152/ajpheart.00187.2018"><em>American Journal of Heart and Circulation Physiology</em>, article in press</a></li>
<li>Sarah Kim-Hellmuth, Matthias Bechheim, Benno Puetz, Pejman Mohammadi, Yohann Nedelec, <strong>Nicholas Giangreco</strong>, Jessica Becker, Vera Kaiser, Nadine Fricker, Esther Beier, Peter Boor, Stephane Castel, Markus M. Noethen, Luis B. Barreiro, Joseph K. Pickrell, Bertram Mueller-Myhsok, Tuuli Lappalainen, Johannes Schumacher, Veit Hornung. Genetic regulatory effects modified by immune activation contribute to autoimmune disease associations <a href="https://www.nature.com/articles/s41467-017-00366-1"><em>Nature Communications</em>, 8 (266): 1-10</a>.</li>
<li><strong>Giangreco N.</strong>, Petrykowska H., Scott A., Margolin G., Gotea V., Cho K. R., and Elnitski L. Inactivation of <em>Arid1a</em> drives aberrant epigenetic traits in a mouse model of <em>Apc</em>/<em>Pten</em> defective ovarian endometrioid tumors. <em>(in preparation)</em></li>
</ul>
<p><a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=Giangreco+N%5BAuthor%5D"><strong>Peer-Reviewed Publications on Pubmed</strong></a></p>
<h2 id="posters-and-software">POSTERS AND SOFTWARE</h2>
<ul>
<li><strong>Nick Giangreco</strong> and Nicholas Tatonetti. Using precision pharmacovigilance to detect developmentally-regulated adverse drug reactions: a case study with antiepileptic drugs. <a href="https://doi.org/10.7490/f1000research.1116203.1">poster</a> <a href="https://github.com/ngiangre/pediatric_pharmacovigilance">github</a></li>
<li><strong>Nick Giangreco</strong> and Nicholas Tatonetti. Using precision pharmacovigilance to detect and evaluate antiepileptic drug associated adverse reactions in pediatric patients. <a href="https://f1000research.com/posters/7-1091">poster</a></li>
<li><strong>Nick Giangreco</strong> and Nicholas Tatonetti. cohorts. <a href="https://github.com/ngiangre/cohorts">github</a></li>
<li><strong>Nick Giangreco</strong>. Scan2CNV. <a href="https://omictools.com/scan2cnv-tool">OMICSTools</a></li>
<li><strong>Giangreco N</strong>, Zorn E, Chen E et al. Identification of novel primary graft dysfunction biomarkers using exosome proteomics [version 1; not peer reviewed]. F1000Research 2017, 6:2080 <a href="https://f1000research.com/posters/6-2080">(poster)</a> (doi: 10.7490/f1000research.1115115.1)</li>
<li><strong>Giangreco N</strong> and Lezon T. Alternative conformation prediction of Vibrio Cholerae concentrative nucleoside transporter. F1000Posters 2013, 4:776 <a href="https://f1000research.com/posters/1093921">(poster)</a>.</li>
</ul>
<h2 id="leadership-and-management-experience">Leadership and Management Experience</h2>
<ul>
<li><a href="https://sites.google.com/view/cumcdatasciencegroup/home">CUIMC Data Science Club</a>
<ul>
<li>President</li>
<li>Organize and manage team of officers, oversee activities, manage budget, and promote outreach portfolio to support and strengthen the data science skills of biomedical scientists at CUIMC</li>
</ul></li>
<li><a href="http://www.healthtechassembly.com/">Health Tech Assembly</a>
<ul>
<li>President
<ul>
<li>Manage business, medical, public health, and engineering representatives for inter-school events, panels, and conferences at Columbia.</li>
<li>Organize and plan professional/social engagement events.</li>
<li>Network and connect with NYC-wide entrepreneurs and professionals.</li>
</ul></li>
</ul></li>
<li><a href="https://www.gsas.cuimc.columbia.edu/student-life/graduate-student-organization">Graduate Student Organization at Columbia University Irving Medical Center</a>
<ul>
<li>Co-President
<ul>
<li>Organize and manage team of Biomedical PhDs for social and professional activities serving hundreds of CUIMC PhD students.</li>
</ul></li>
</ul></li>
<li><a href="https://www.cc-seas.columbia.edu/student-group/columbia-graduate-council">Columbia Graduate Council</a>
<ul>
<li>Former Treasurer
<ul>
<li>Establish budget and expense sheets and annual reports</li>
<li>Manage forty thousand dollar budget for Columbia inter-school activities.</li>
</ul></li>
</ul></li>
<li>NYC MeetUps
<ul>
<li><a href="https://www.meetup.com/NYHAIS/">New York Health Artificial Intelligence Society</a>
<ul>
<li>Co-organizer</li>
<li>Promote public discourse on a wide range of topics such as AI &amp; Society, AI &amp; Healthcare, and economic impact by AI.</li>
<li>Facilitate group engagement, AI study groups, and group organization.</li>
<li>Consultant on data science and education projects and initiatives.</li>
</ul></li>
</ul></li>
<li>University of Rochester Alumni Undergraduate Interviewer
<ul>
<li>Assess potential and suitability for undergraduate college</li>
<li>Interviewer at large and small event settings</li>
</ul></li>
</ul>
<h2 id="mentoring-tutoring-and-writing">MENTORING, TUTORING, and WRITING</h2>
<ul>
<li><a href="https://www.eventbrite.com/e/healthcare-hack-night-tickets-72013860395">“Hack nights – Solving healthcare data-science/AI/ML problems”</a> Introduction to cancer genomics three part series. Co-led by Nicholas Giangreco and Matthew Eng</li>
<li>Nicholas Giangreco. <a href="http://www.phdish.com/blog/openscience">“The Importance of being Open”</a>. <em>PHDISH</em> January 9th 2019.</li>
<li>Undergraduate Mentoring:
<ul>
<li>Payal Chandak, Columbia University
<ul>
<li>Provide guidance and instruction in biomedical data science and research training.</li>
<li>Provided guidance and mentoring for summer 2018 research internship in the Tatonetti Lab
<ul>
<li><a href="https://github.com/tatonetti-lab/sex_risks">“Drugs with sex-linked risk for adverse drug reactions”</a></li>
</ul></li>
</ul></li>
</ul></li>
<li>Curiology tutor
<ul>
<li>Managed and co-led science experiments with NIH fellows for middle school students in Washington D.C.</li>
</ul></li>
<li>College Bound tutor
<ul>
<li>Facilitated completion of homework assignments in STEM for Washington D.C. high school students.</li>
</ul></li>
<li>Genetics Study Group Leader, Center for Excellence in Teaching and Learning, University of Rochester.</li>
</ul>
<h2 id="conferences-and-hackathons">CONFERENCES AND HACKATHONS</h2>
<ul>
<li><p><a href="https://meetings.cshl.edu/meetings.aspx?meet=data&amp;year=20">CSHL Biological Data Science meeting November 2018.</a></p></li>
<li><p><a href="https://www.ohdsi.org/2018-ohdsi-symposium/">OHDSI 2018 Symposium</a></p></li>
<li><a href="https://biohackathons.github.io/">NCBI hackathon @ New York Genome Center August 2018</a>
<ul>
<li>Project lead for developing data science notebooks and web application interfacing with drug safety data.</li>
<li>See <a href="https://github.com/NCBI-Hackathons/SafeDrugs">SafeDrugs</a> github repository.</li>
</ul></li>
<li><p><a href="https://iscb.org/ismb2018">Intelligent Systems in Molecular Biology, July 2018</a></p></li>
<li><p><a href="http://nechs.org/">Second Northeast Computational Health Summit, April 2018</a></p></li>
<li><p>American Heart Association Scientific Sessions 2017, poster presentation <a href="https://f1000research.com/posters/6-2080"><em>Giangreco et al. 2017</em></a>.</p></li>
<li><a href="https://github.com/NCBI-Hackathons/Proteomic_Correlation_Shiny">NCBI Hackathon @ New York Genome Center June 2017.</a></li>
<li><a href="https://github.com/NCBI-Hackathons/Scan2CNV">NCBI Hackathon @ NCBI March 2017.</a></li>
<li>CSHL Biological Data Science meeting October 2016.</li>
<li><a href="https://github.com/NCBI-Hackathons/DASH_cell_type">JHU DaSH Hackathon September 2015.</a></li>
<li><p>ISMB/ECCB conference @ Berlin, Germany July 2013, poster presentation <em>Giangreco et al. 2013</em>.</p></li>
</ul>
<h2 id="talks-and-panels">TALKS AND PANELS</h2>
<ul>
<li>“Intro to Bioinformatics and How to Analyze Brain Tissue with Data Science”. Invited presentation at <a href="https://www.meetup.com/NYC-Medical-Research-and-Bioinformatics-Group/">NYC Medical Research and Bioinformatics Group</a>. November 2018. <a href="https://github.com/ngiangre/Miscellaneous_but_Useful/blob/master/20181113_Bioinformatics_Meetup_Presentation.pdf">Presentation link</a>.</li>
<li>Medical Research Career Panelist, Minds Matter NYC, June 2018</li>
<li>Standardized and Reproducible Analysis Enables Identification of Novel Primary Graft Dysfunction Biomarkers using Exosome Proteomics, <a href="http://nechs.org">Second NorthEast Computational Health Summit 2018</a>, April 2018.</li>
<li><a href="http://nickg.bio/_posts/20171220_NYHAIS_talk.pdf">“Tools, Libraries and Analyses in Biomedical Data Science”</a>, New York Healthcare Artificial Intelligence Society, December 2017.</li>
<li><a href="http://nickg.bio/_posts/20171205_LateNightScience_Talk.pdf">“Doing Science with Big Data”</a>, Late Night Science, Columbia University Neuroscience Outreach, December 2017.</li>
<li>“AI, Life Sciences, and Big Data”, New York Healthcare Artificial Intelligence Society, August 2017. <a href="https://github.com/ngiangre/20171220_AI_MeetUp_Presentation">Presentation link</a>.</li>
<li>NIDDK Undergraduate Step-Up Judge, NIH, Bethesda MD, August 2015.</li>
</ul>
<h2 id="professional-memberships">PROFESSIONAL MEMBERSHIPS</h2>
<ul>
<li>American Heart Association, 2017-.</li>
<li>American Medical Informatics Association, 2017-.</li>
<li>International Society of Computational Biology, 2013-2014 &amp; 2017-.</li>
</ul>
</body>
</html>
