<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>Nick Giangreco</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    .display.math{display: block; text-align: center; margin: 0.5rem auto;}
  </style>
  <link rel="stylesheet" href="cv_style.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Nick Giangreco</h1>
</header>
<!--  

Great resource https://blog.chmd.fr/editing-a-cv-in-markdown-with-pandoc.html 

The CSS is attributable to the above blog post

This is another great post influencing this document https://mszep.github.io/pandoc_resume/

see github repo for licensing information

-->
<h3 id="independent-and-collaborative-precision-medicine-scientist-and-engineer"><em>Independent and collaborative precision medicine scientist and engineer</em></h3>
<p><a href="http://github.com/ngiangre">Github: ngiangre</a> | <a href="http://www.linkedin.com/in/nickgiangreco/">Linkedin: http://www.linkedin.com/in/nickgiangreco/</a> | <a href="https://orcid.org/0000-0001-8138-4947">ORCID: 0000-0001-8138-4947</a> | <a href="http://nickg.bio">Personal Website: nickg.bio</a> | <a href="nick.giangreco@gmail.com">Email: nick.giangreco@gmail.com</a> | <strong>Date of preparation:</strong> April 26th 2025</p>
<table>
<colgroup>
<col style="width: 100%" />
</colgroup>
<tbody>
<tr class="odd">
<td style="text-align: left;"><strong>Topic expertise:</strong> Pediatric drug safety, Bioinformatics, Biomarker discovery, Systems Biology, Statistics, Interpretable machine learning and artificial intelligence</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong>Technical expertise:</strong> Biomedical data science, (R/Python) software package development, R programming, {plumber} APIs, machine learning, statistical simulation, data cleaning, data pipelines, Shiny (for R and Python) application development, object oriented programming, RMarkdown/JuPyteR/Quarto, Microsoft Power App and API user</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong>Programming Languages:</strong> R, Python, SQL, HTML, CSS, Observable Javascript, Java, Bash</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong>Patents:</strong> <a href="https://patents.google.com/patent/WO2022060842A1/en?oq=2022060842">Systems and methods for predicting graft dysfunction with exosome proteins</a></td>
</tr>
</tbody>
</table>
<h2 id="work-experience">WORK EXPERIENCE</h2>
<dl>
<dt>Feb 2023 - Present</dt>
<dd><p><strong>Computer Sciences Advisor (Remote)</strong> <a href="http://www.mindarchhealth.com/">(MindArch Health)</a>; Sayville, NY</p>
<p>Machine learning, Software, Web Applications &amp; Design</p>
</dd>
<dt>Dec 2021 - Present</dt>
<dd><p><strong>Senior Scientist, Quantitative Translational Science (Remote)</strong> <a href="https://www.regeneron.com">(Regeneron Pharmaceuticals Inc.)</a>; Tarrytown, NY</p>
<p>Precision Medicine - scientist, data architect, software developer, and applications engineer</p>
<ul>
<li>Analyze clinical and biomarker data in rare disease and drug trials</li>
<li>Perform and manage data architecture and software development projects in precision medicine innovation development portfolio.</li>
<li>Develop and maintain 6+ software packages and data pipelines for team of clinical biomarker data analysts.</li>
<li>Deliver automated project management and portfolio reports to senior leadership.</li>
<li>Mentor interns and data scientists to develop reproducible data analyses and web apps.</li>
</ul>
</dd>
<dt>Aug 2016 - Oct 2021</dt>
<dd><p><strong>Systems biologist</strong> <a href="https://systemsbiology.columbia.edu/people/nicholas-giangreco">(Columbia University)</a>; New York, NY</p>
<ul>
<li>Ph.D. Thesis: <a href="https://doi.org/10.7916/d8-5d9b-6738">Mind the developmental gap: Identifying adverse drug effects across childhood to evaluate biological mechanisms from growth and development</a></li>
<li><a href="https://www.cell.com/med/issue?pii=S2666-6340(21)X0020-7#fullCover">Cell Press cover art</a></li>
<li><a href="https://pdsportal.shinyapps.io/pdsportal/">PDSportal Shiny for R application</a></li>
<li><a href="https://github.com/ngiangre/kidsides">kidsides R package and database</a></li>
<li><a href="https://www.dbmi.columbia.edu/pediatric-drug-side-effects-study-tatonetti/">Columbia University media release</a></li>
</ul>
</dd>
<dt>Feb 2021 - Aug 2021</dt>
<dd><p><strong>Bioinformatics intern</strong> <a href="https://www.dnanexus.com/">(DNAnexus)</a>; San Francisco, CA</p>
<p>Solutions Science team</p>
<p>Initiated internal and external projects such as interpretable, tree-based andd boosting machine learning for phenotype prediction and 2) integration of genomic/phenomic data using common data models</p>
</dd>
<dt>Jun 2019 - Aug 2019</dt>
<dd><p><strong>Clinical informatics intern</strong> <a href="https://www.regeneron.com/genetics-center">(Regeneron Genetics Center)</a>); Tarrytown, NY</p>
<p>Developed database of multivariate clinical associations using incremental learning technique on EC2 Amazon Web Services.</p>
</dd>
<dt>Jul 2018 - Sept 2018</dt>
<dd><p><strong>Computational biology intern</strong> <a href="https://www.genetic-leap.com">(Genetic Leap)</a>; New York, NY</p>
<p>Formerly Genetic Intelligence Inc.</p>
<p>Conducted independent and collaborative genomics research using NCBI APIs and Amazon Web Services.</p>
</dd>
<dt>Aug 2014 - July 2019</dt>
<dd><p><strong>Cancer bioinformatician</strong> <a href="https://www.genome.gov/staff/Laura-Elnitski-PhD">(National Human Genome Research Institute)</a>; Bethesda, MD</p>
<p>Post-baccalaureate trainee 2014-2016; Special volunteer 2016-2019</p>
<p>Investigated ovarian endometrioid tumorigenesis by integrating and analyzing RNA-Seq and DNA methylation sequencing (MBD-Seq).</p>
</dd>
</dl>
<h2 id="education">EDUCATION</h2>
<dl>
<dt>2016 - 2021</dt>
<dd><p><a href="http://systemsbiology.columbia.edu/people/nicholas-giangreco"><strong>PhD, Systems Biology</strong></a>; Columbia University, New York City, NY</p>
<p>Thesis Date: October 26th, 2021</p>
<p>Department of Systems Biology and Biomedical Informatics</p>
<p>PhD advisor: <a href="http://tatonettilab.org">Dr. Nicholas Tatonetti</a></p>
<p>Masters of Arts (2018) and Masters of Philosophy (2019)</p>
</dd>
<dt>2010 - 2014</dt>
<dd><strong>BS, Biochemistry</strong>; University of Rochester, Rochester, NY
</dd>
</dl>
<h2 id="select-publications">SELECT PUBLICATIONS</h2>
<p>A complete listing may be found in <a href="https://pubmed.ncbi.nlm.nih.gov/?term=Giangreco+N&amp;cauthor_id=28814792"><strong>pubmed</strong></a> or <a href="https://scholar.google.com/citations?user=Q3m0Jl8AAAAJ&amp;hl=en&amp;oi=ao"><strong>google scholar</strong></a></p>
<ol type="1">
<li>Biswas S, Shahriar S, <strong>Giangreco NP</strong>, Arvanitis P, Winkler M, Tatonetti NP, Brunken WJ, Cutforth T, Agalliu D. Mural Wnt/β-catenin signaling regulates Lama2 expression to promote neurovascular unit maturation. Development. 2022 Sep 1;149(17):dev200610. doi: <a href="https://doi.org/10.1242/dev.200610">10.1242/dev.200610</a>. Epub 2022 Sep 13. PMID: 36098369; PMCID: PMC9578690.</li>
<li><strong>Giangreco, N.P.</strong>, Lebreton, G., Restaino, S. et al. Alterations in the kallikrein-kinin system predict death after heart transplant. Sci Rep 12, 14167 (2022). <a href="https://doi.org/10.1038/s41598-022-18573-2">https://doi.org/10.1038/s41598-022-18573-2</a></li>
<li><strong>Giangreco NP</strong>, Tatonetti NP. A database of pediatric drug effects to evaluate ontogenic mechanisms from child growth and development. Med (N Y). 2022 Aug 12;3(8):579-595.e7. doi: <a href="https://doi.org/10.1016/j.medj.2022.06.001">10.1016/j.medj.2022.06.001</a>. Epub 2022 Jun 24. PMID: 35752163; PMCID: PMC9378670.</li>
<li><strong>Giangreco, Nicholas</strong> (2022), Longitudinal trends of EHR concepts in pediatric patients, Dryad, Dataset, <a href="https://doi.org/10.5061/dryad.j0zpc86g3">https://doi.org/10.5061/dryad.j0zpc86g3</a></li>
<li><strong>Nicholas P Giangreco</strong>, Sulieman Lina, Jun Qian, Aymone Kuoame, Vignesh Subbian, Eric Boerwinkle, Mine Cicek, Cheryl R Clark, Elizabeth Cohen, Kelly A Gebo, Roxana Loperena-Cortes, Kelsey Mayo, Stephen Mockrin, Lucila Ohno-Machado, Sheri D Schully, Nicholas P Tatonetti, Andrea H Ramirez, Pediatric data from the All of Us research program: demonstration of pediatric obesity over time, JAMIA Open, Volume 4, Issue 4, October 2021, ooab112, <a href="https://doi.org/10.1093/jamiaopen/ooab112">https://doi.org/10.1093/jamiaopen/ooab112</a></li>
<li><strong>Giangreco NP</strong>, Lebreton G, Restaino S, Jane Farr M, Zorn E, Colombo PC, Patel J, Levine R, Truby L, Soni RK, Leprince P, Kobashigawa J, Tatonetti NP, Fine BM. Plasma kallikrein predicts primary graft dysfunction after heart transplant. J Heart Lung Transplant. 2021 Jul 10:S1053-2498(21)02391-3. doi: <a href="https://doi.org/10.1016/j.healun.2021.07.001">10.1016/j.healun.2021.07.001</a>.</li>
<li><strong>Giangreco, N.P.</strong>, Tatonetti, N.P. Evaluating risk detection methods to uncover ontogenic-mediated adverse drug effect mechanisms in children. BioData Mining 14, 34 (2021). <a href="https://doi.org/10.1186/s13040-021-00264-9">https://doi.org/10.1186/s13040-021-00264-9</a>.</li>
<li><strong>Giangreco, NP</strong>, Elias, JE, Tatonetti, NP. No population left behind: Improving paediatric drug safety using informatics and systems biology. Br J Clin Pharmacol. 2021; 1– 7. <a href="https://doi.org/10.1111/bcp.14705">https://doi.org/10.1111/bcp.14705</a>.</li>
<li><strong>Nicholas P. Giangreco</strong>, Barry Fine, Nicholas P. Tatonetti. cohorts: A Python package for clinical ‘omics data management. bioaRxiv doi: <a href="https://www.biorxiv.org/content/10.1101/626051">https://www.biorxiv.org/content/10.1101/626051</a></li>
<li>Benjamin S Glicksberg, Boris Oskotsky, Phyllis M Thangaraj, <strong>Nicholas Giangreco</strong>, Marcus A Badgeley, Kipp W Johnson, Debajyoti Datta, Vivek A Rudrapatna, Nadav Rappoport, Mark M Shervey, Riccardo Miotto, Theodore C Goldstein, Eugenia Rutenberg, Remi Frazier, Nelson Lee, Sharat Israni, Rick Larsen, Bethany Percha, Li Li, Joel T Dudley, Nicholas P Tatonetti, Atul J Butte, PatientExploreR: an extensible application for dynamic visualization of patient clinical history from electronic health records in the OMOP common data model, Bioinformatics, Volume 35, Issue 21, 1 November 2019, Pages 4515–4518, <a href="https://doi.org/10.1093/bioinformatics/btz409">https://doi.org/10.1093/bioinformatics/btz409</a>.</li>
<li>Benjamin S Glicksberg, Boris Oskotsky, <strong>Nicholas Giangreco</strong>, Phyllis M Thangaraj, Vivek Rudrapatna, Debajyoti Datta, Remi Frazier, Nelson Lee, Rick Larsen, Nicholas P Tatonetti, Atul J Butte, ROMOP: a light-weight R package for interfacing with OMOP-formatted electronic health record data, JAMIA Open, Volume 2, Issue 1, April 2019, Pages 10–14, <a href="https://doi.org/10.1093/jamiaopen/ooy059">https://doi.org/10.1093/jamiaopen/ooy059</a></li>
<li>Estibaliz Castillero, Ziad A. Ali, Hirokazu Akashi, <strong>Nicholas Giangreco</strong>, Catherine Wang, Eric J. Stöhr, Ruping Ji, Xiaokan Zhang, Nathaniel Kheysin, Joo-Eun S. Park, Sheetal Hegde, Sanatkumar Patel, Samantha Stein, Carlos Cuenca, Diana Leung, Shunichi Homma, Nicholas P. Tatonetti, Veli K. Topkara, Koji Takeda, Paolo C. Colombo, Yoshifumi Naka, H. Lee Sweeney, P. Christian Schulze, and Isaac George American Journal of Physiology-Heart and Circulatory Physiology 2018 315:5, <a href="https://www.physiology.org/doi/10.1152/ajpheart.00187.2018">H1463-H1476</a>.</li>
<li>Kim-Hellmuth, S., Bechheim, M., Pütz, B., Mohammadi, P., Néd´lec, Y, <strong>Giangreco, N.</strong>, et al. Genetic regulatory effects modified by immune activation contribute to autoimmune disease associations. Nat Commun 8, 266 (2017). <a href="https://doi.org/10.1038/s41467-017-00366-1">https://doi.org/10.1038/s41467-017-00366-1</a>.</li>
</ol>
<h2 id="fellowships-and-awards">FELLOWSHIPS AND AWARDS</h2>
<ul>
<li><p>Travel award to present on a panel at the 2022 American Medical Informatics Association Summit in Chicago, USA</p></li>
<li><p>Travel award to 2021 Elixir biohackathon in Barcelona Spain.</p></li>
<li><p>Special recognition from Columbia University for service during the COVID-19 crisis, 2021</p></li>
<li><p>2021 Diversity &amp; Inclusion Commercialization and Entrepreneurship Fellow @ Columbia Technology Ventures</p></li>
<li><p>Three-Minute Thesis 2019 finalist @ Columbia Graduate School of Arts and Sciences.</p></li>
<li><p>Best contribution in methodological research at the <a href="https://www.ohdsi.org/2018-ohdsi-symposium/">OHDSI 2018 Symposium</a> for <a href="http://10.7490/f1000research.1116203.1">Pediatric Drug Safety</a> poster.</p></li>
<li><p>Columbia Diversity Fellowship 2016.</p></li>
<li><p>Department of Systems Biology Merit Fellowship 2016.</p></li>
<li><p>Donald Charles Award, University of Rochester Department of Biology, 2014.</p></li>
<li><p>Fulbright Fellowship Alternate 2013-2014: Sweden, Molecular Modeling, “Novel Antibody-SpA Complex Modeling”.</p></li>
<li><p>Travel Award to 9<sup>th</sup> Student Council and ISMB/ECCB conference 2013 Berlin, Germany.</p></li>
</ul>
<h2 id="posters-and-software">POSTERS AND SOFTWARE</h2>
<ul>
<li><strong>Nicholas Giangreco</strong>, Salvatore G. Volpe, Meghana Tandon, Kamileh Narisnh, Ben Busby. All Genes Lead to ROMOPOmics. <a href="https://ohdsi.org/2021-global-symposium-showcase-77/">OHDSI symposium video demo</a></li>
<li><strong>Nick Giangreco</strong> and Nicholas Tatonetti. Using precision pharmacovigilance to detect developmentally-regulated adverse drug reactions: a case study with antiepileptic drugs. <a href="https://doi.org/10.7490/f1000research.1116203.1">poster</a> <a href="https://github.com/ngiangre/pediatric_pharmacovigilance">github</a></li>
<li><strong>Nick Giangreco</strong> and Nicholas Tatonetti. Using precision pharmacovigilance to detect and evaluate antiepileptic drug associated adverse reactions in pediatric patients. <a href="https://f1000research.com/posters/7-1091">poster</a></li>
<li><strong>Nick Giangreco</strong> and Nicholas Tatonetti. cohorts. <a href="https://github.com/ngiangre/cohorts">github</a></li>
<li><strong>Nick Giangreco</strong>. Scan2CNV. <a href="https://omictools.com/scan2cnv-tool">OMICSTools</a></li>
<li><strong>Giangreco N</strong>, Zorn E, Chen E et al. Identification of novel primary graft dysfunction biomarkers using exosome proteomics [version 1; not peer reviewed]. F1000Research 2017, 6:2080 <a href="https://f1000research.com/posters/6-2080">(poster)</a> (doi: 10.7490/f1000research.1115115.1)</li>
<li><strong>Giangreco N</strong> and Lezon T. Alternative conformation prediction of Vibrio Cholerae concentrative nucleoside transporter. F1000Posters 2013, 4:776 <a href="https://f1000research.com/posters/1093921">(poster)</a>.</li>
</ul>
<h2 id="leadership-and-management-experience">Leadership and Management Experience</h2>
<ul>
<li>University of Rochester Alumni Buffalo Leadership Council 2023-Present
<ul>
<li>Ideate and organize local events</li>
</ul></li>
<li><a href="https://www.nyhais.org">New York Health Artificial Intelligence Society</a> 2019-2024
<ul>
<li>501(c)(3) not-for-profit Cofounder and Secretary, 2019-2024</li>
<li>Promote public discourse on a wide range of topics such as AI &amp; Society, AI &amp; Healthcare, and economic impact by AI.</li>
<li>Organize and facilitate group engagement, workshops, AI study groups, and not-for-profit organization.</li>
<li>Consultant on data science and education projects and initiatives.</li>
</ul></li>
<li>University of Rochester Alumni Undergraduate Interviewer 2016-2022
<ul>
<li>Assess potential and suitability for undergraduate college</li>
<li>Interviewer at large and small event settings</li>
</ul></li>
<li><a href="https://github.com/adriennepichon/DBMI_JusticeInformatics">CUIMC Department of Biomedical Informatics Justice Informatics Reading group</a>
<ul>
<li>Member January 2021-December 2021</li>
<li>Read and discuss publications, books, and policy on fairness and ethics within informatics development and their application</li>
</ul></li>
<li><a href="https://sites.google.com/view/cumcdatasciencegroup/home">CUIMC Data Science Club</a>
<ul>
<li>President 2017-2021</li>
<li>Organize and manage team of officers, oversee activities, manage budget, and promote outreach portfolio to support and strengthen the data science skills of biomedical scientists at CUIMC</li>
</ul></li>
<li><a href="http://www.healthtechassembly.com/">Health Tech Assembly</a>
<ul>
<li>President 2019-2020
<ul>
<li>Manage business, medical, public health, and engineering representatives for inter-school events, panels, and conferences at Columbia.</li>
<li>Organize and plan professional/social engagement events.</li>
<li>Network and connect with NYC-wide entrepreneurs and professionals.</li>
</ul></li>
<li>Medical campus representative 2018-2019</li>
</ul></li>
<li><a href="https://www.gsas.cuimc.columbia.edu/student-life/graduate-student-organization">Graduate Student Organization at Columbia University Irving Medical Center</a>
<ul>
<li>Co-President 2019-2020
<ul>
<li>Organize and manage team of Biomedical PhDs for social and professional activities serving hundreds of CUIMC PhD students.</li>
</ul></li>
<li>Finance Chair 2018-2019</li>
</ul></li>
<li><a href="https://www.cc-seas.columbia.edu/student-group/columbia-graduate-council">Columbia Graduate Council</a>
<ul>
<li>Treasurer 2019-2020
<ul>
<li>Establish budget and expense sheets and annual reports</li>
<li>Manage forty thousand dollar budget for Columbia inter-school activities.</li>
</ul></li>
</ul></li>
<li><a href="https://www.dbmi.columbia.edu/">Department of Biomedical Informatics</a>
<ul>
<li>Co-lead weekly seminar series</li>
<li>Manage presentation schedule, speaker logistics and travel, and promote team coordination.</li>
</ul></li>
<li><a href="https://systemsbiology.columbia.edu/">Department of Systems Biology</a>
<ul>
<li>Point person for Systems Biology Trainee Council</li>
<li>Secured funding and launched event portfolio for trainee development</li>
<li>Manage monthly departmental happy hours</li>
</ul></li>
</ul>
<h2 id="mentoring-tutoring-and-writing">MENTORING, TUTORING, and WRITING</h2>
<ul>
<li>Mentor in the <a href="https://sayyesbuffalo.org">Say Yes not-for-profit</a> in Buffalo NY
<ul>
<li>Mission is to remove barriers to educational and economic attainment for our young people.</li>
</ul></li>
<li>Mentor in technical and entrepreneurship development for the <a href="https://www.innovatechildrenshealth.com/">Innovate Children’s Health Challenge</a></li>
<li>Advisor, mentor and tutor for project management, R and python programming, and statistics and machine learning to high school/college/graduate students and professionals.</li>
<li><a href="https://www.eventbrite.com/e/healthcare-hack-night-tickets-72013860395">“Hack nights – Solving healthcare data-science/AI/ML problems”</a> Introduction to cancer genomics four part series. Co-led with Matthew Eng
<ul>
<li><a href="https://medium.com/adventures-in-healthcare-data">Adventures In Hacking Healthcare Medium Publication</a></li>
</ul></li>
<li>Nicholas Giangreco. <a href="http://www.phdish.com/blog/openscience">“The Importance of being Open”</a>. <em>PHDISH</em> January 9th 2019.</li>
<li>Mentoring:
<ul>
<li><a href="https://www.linkedin.com/in/payal-chandak-366403132/">Payal Chandak</a>, Undergraduate at Columbia University
<ul>
<li>Provide guidance and instruction in biomedical data science and research training.</li>
<li>Provided guidance and mentoring for summer 2018 research internship in the Tatonetti Lab
<ul>
<li><a href="https://github.com/tatonetti-lab/sex_risks">“Drugs with sex-linked risk for adverse drug reactions”</a></li>
</ul></li>
</ul></li>
<li>SMRI high school mentorship
<ul>
<li>Provide guidance to high school student in <a href="https://github.com/veronikabreton/smri2020">biomedical data science research project</a></li>
</ul></li>
</ul></li>
<li>Curiology tutor
<ul>
<li>Managed and co-led science experiments with NIH fellows for middle school students in Washington D.C.</li>
</ul></li>
<li>College Bound tutor
<ul>
<li>Facilitated completion of homework assignments in STEM for Washington D.C. high school students.</li>
</ul></li>
<li>Genetics Study Group Leader, Center for Excellence in Teaching and Learning, University of Rochester.</li>
</ul>
<h2 id="conferences-and-hackathons">CONFERENCES AND HACKATHONS</h2>
<ul>
<li><p><a href="https://www.rstudio.com/conference/">posit::conf( c(2023,2024) ) with workshops</a></p></li>
<li><p><a href="https://www.bio-itworldexpo.com/">Bio-IT World 2023</a></p></li>
<li><p><a href="https://www.rstudio.com/conference/">rstudio::conf(2022)</a></p></li>
<li><p><a href="https://amia.org/education-events/amia-2022-informatics-summit">American Medical Informatics Association Translational Summit 2022 Chicago USA</a></p></li>
<li><p><a href="https://biohackathon-europe.org/">2021 Elixir biohackathon</a></p>
<ul>
<li>Front-end developer</li>
<li>Built prototype dashboard for improved disease subtyping and treatment pathways for colorectal cancer. See <a href="https://github.com/collaborativebioinformatics/CRC-Dashboard">github</a> and <a href="https://nick-giangreco.shinyapps.io/cds-dashboard/">shiny dashboard</a>.</li>
</ul></li>
<li><p><a href="https://www.library.cmu.edu/about/publications/news/upcoming-event-hackathon-2021">Clinical Reporting of Multi ’Omics data</a></p>
<ul>
<li>Led team and managed hackathon teams to manage and streamline integration of genomic, transcriptomic, and polygenic risk score data into the OMOP common data model. See <a href="https://github.com/collaborativebioinformatics/omics_to_omop">github</a>.</li>
</ul></li>
<li><p><a href="https://github.com/elixir-europe/BioHackathon-projects-2020">Elixir biohackathon</a></p>
<ul>
<li>Collaborated with bioinformatics team to integrate nextflow scheme and cancer mutation data (vcf files) into OMOP standard structure using <a href="https://github.com/AndrewC160/ROMOPOmics">ROMOPOmics</a>.</li>
</ul></li>
<li><p><a href="https://nfhack-platform.bemyapp.com/#/event">Hack for NF, Children’s Tumor Foundation</a></p>
<ul>
<li>Co-lead web developer coordination and product development</li>
<li>A FHIR-complient and primarily patient-centric profile for NF management and centralized repository for medical journey.</li>
</ul></li>
<li><p>Judge, Predictive analytics track, Columbia University COVID-19 Data Challenge</p></li>
<li><p><a href="https://interoperabilityinstitute.org/virtual-interopathon/">Virtual Interoperathon 2020</a></p>
<ul>
<li>Project developer for personalized health record accessed with fhir-client python api</li>
<li>Developed flask application <a href="https://github.com/ngiangre/District-13_Sandbox">proof-of-concept</a>.</li>
</ul></li>
<li><p><a href="https://biohackathons.github.io/">NCBI Hackathon @ Carnegie Mellon University January 2020</a></p>
<ul>
<li>Project co-lead for developing and extending common data model to represent biological ’omics data for reproducible queries and analyses.</li>
<li>See original <a href="https://github.com/NCBI-Codeathons/OMOPOmics">OMOPOmics</a> and R package <a href="https://github.com/AndrewC160/ROMOPOmics">ROMOPOmics</a> github repository.</li>
</ul></li>
<li><p><a href="https://meetings.cshl.edu/meetings.aspx?meet=data&amp;year=20">CSHL Biological Data Science meeting November 2018.</a></p></li>
<li><p><a href="https://www.ohdsi.org/2018-ohdsi-symposium/">OHDSI 2018 Symposium</a></p></li>
<li><p><a href="https://biohackathons.github.io/">NCBI Hackathon @ New York Genome Center August 2018</a></p>
<ul>
<li>Project lead for developing data science notebooks and web application interfacing with drug safety data.</li>
<li>See <a href="https://github.com/NCBI-Hackathons/SafeDrugs">SafeDrugs</a> github repository.</li>
</ul></li>
<li><p><a href="https://iscb.org/ismb2018">Intelligent Systems in Molecular Biology, July 2018</a></p></li>
<li><p><a href="http://nechs.org/">Second Northeast Computational Health Summit, April 2018</a></p></li>
<li><p>American Heart Association Scientific Sessions 2017, poster presentation <a href="https://f1000research.com/posters/6-2080"><em>Giangreco et al. 2017</em></a>.</p></li>
<li><p><a href="https://github.com/NCBI-Hackathons/Proteomic_Correlation_Shiny">NCBI Hackathon @ New York Genome Center June 2017.</a></p></li>
<li><p><a href="https://github.com/NCBI-Hackathons/Scan2CNV">NCBI Hackathon @ NCBI March 2017.</a></p></li>
<li><p>CSHL Biological Data Science meeting October 2016.</p></li>
<li><p><a href="https://github.com/NCBI-Hackathons/DASH_cell_type">JHU DaSH Hackathon September 2015.</a></p></li>
<li><p>ISMB/ECCB conference @ Berlin, Germany July 2013, poster presentation <em>Giangreco et al. 2013</em>.</p></li>
</ul>
<h2 id="talks-and-panels">TALKS AND PANELS</h2>
<ul>
<li>Panelist for S14: “Utility of the All of Us Researcher Workbench in Educational and Research Settings” <a href="https://amia.org/education-events/amia-2022-informatics-summit">American Medical Informatics Association Informatics Summit</a> March 2022 Chicago.</li>
<li>DBMI Justice Informatics Joint Forum. Internal, collaborative meeting discussing the impact of Big Data and mathematical models on inequity and how that applies to research.</li>
<li>“Pediatrics data in <em>All of Us</em>”. All of Us Research Workbench Onramp Virtual Event June 2021.</li>
<li>2021 Graduate Scholar for research talk entitled ‘Mind the developmental gap: Identifying adverse drug effects across childhood to evaluate biological mechanisms from growth and development’ with <a href="https://www.youtube.com/channel/UCf6KOt3vTnXZQua94atQ1Dg">Emuritus Professors at Columbia (EPIC)</a></li>
<li>“Understanding dynamics with statistical modeling” Coding workshop, CUIMC Data Science Club, <a href="https://www.youtube.com/watch?v=ObmdJ-1Ok_Q">YouTube video</a></li>
<li>“Introduction to programming and bioinformatics” Tutorial for 2021 Columbia University Science Matters Research Internship</li>
<li>“Intro to Bioinformatics and How to Analyze Brain Tissue with Data Science”. Invited presentation at <a href="https://www.meetup.com/NYC-Medical-Research-and-Bioinformatics-Group/">NYC Medical Research and Bioinformatics Group</a>. November 2018. <a href="https://github.com/ngiangre/Miscellaneous_but_Useful/blob/master/20181113_Bioinformatics_Meetup_Presentation.pdf">Presentation link</a>.</li>
<li>Medical Research Career Panelist, Minds Matter NYC, June 2018</li>
<li>Standardized and Reproducible Analysis Enables Identification of Novel Primary Graft Dysfunction Biomarkers using Exosome Proteomics, <a href="http://nechs.org">Second NorthEast Computational Health Summit 2018</a>, April 2018.</li>
<li><a href="http://nickg.bio/_posts/20171220_NYHAIS_talk.pdf">“Tools, Libraries and Analyses in Biomedical Data Science”</a>, New York Healthcare Artificial Intelligence Society, December 2017.</li>
<li><a href="http://nickg.bio/_posts/20171205_LateNightScience_Talk.pdf">“Doing Science with Big Data”</a>, Late Night Science, Columbia University Neuroscience Outreach, December 2017.</li>
<li>“AI, Life Sciences, and Big Data”, New York Healthcare Artificial Intelligence Society, August 2017. <a href="https://github.com/ngiangre/20171220_AI_MeetUp_Presentation">Presentation link</a>.</li>
<li>NIDDK Undergraduate Step-Up Judge, NIH, Bethesda MD, August 2015.</li>
</ul>
<h2 id="professional-memberships">PROFESSIONAL MEMBERSHIPS</h2>
<ul>
<li>International Society of Computational Biology, 2013-2014 &amp; 2017-2018</li>
<li>American Heart Association, 2017-2018.</li>
<li>American Medical Informatics Association, 2017-2018.</li>
</ul>
</body>
</html>
